From e2419c698922fb65ef62250a4588104021449b55 Mon Sep 17 00:00:00 2001
From: Zdenek Dohnal <zdohnal@redhat.com>
Date: Fri, 6 Mar 2026 08:31:47 +0100
Subject: [PATCH] tls-gnutls.c: Do not check for errno after I/O operations

Based on gnutls_record_send/recv man pages, we should use the return
value of the functions as indicator what happened in the function and do
not look into errno at all.

Checking the errno value caused infinity loop in cupsd on busy servers
if there were enough connection errors when cupsd wrote the response.

The patch is provided by Paul Zirnik from SUSE - thank you for the
patch!

Fixes #827
---
 cups/tls-gnutls.c | 4 ++--

diff --git a/cups/tls-gnutls.c b/cups/tls-gnutls.c
index 64f2af372..5769d6313 100644
--- a/cups/tls-gnutls.c
+++ b/cups/tls-gnutls.c
@@ -1208,7 +1208,7 @@ _httpTLSRead(http_t *http,		/* I - Connection to server */
 
   result = gnutls_record_recv(http->tls, buf, (size_t)len);
 
-  if (result < 0 && !errno)
+  if (result < 0)
   {
    /*
     * Convert GNU TLS error to errno value...
@@ -1669,7 +1669,7 @@ _httpTLSWrite(http_t     *http,		/* I - Connection to server */
 
   result = gnutls_record_send(http->tls, buf, (size_t)len);
 
-  if (result < 0 && !errno)
+  if (result < 0)
   {
    /*
     * Convert GNU TLS error to errno value...
-- 
2.53.0

